app/bundles/CoreBundle/Shortener/Shortener.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CoreBundle\Shortener;
+
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+
+class Shortener
+{
+    public const SHORTENER_SERVICE = 'shortener_service';
+
+    private CoreParametersHelper $coreParametersHelper;
+
+    /**
+     * @var ShortenerServiceInterface[]
+     */
+    private array $services = [];
+
+    public function __construct(CoreParametersHelper $coreParametersHelper)
+    {
+        $this->coreParametersHelper = $coreParametersHelper;
+    }
+
+    public function addService(string $id, ShortenerServiceInterface $shortener): void
+    {
+        $this->services[$id] = $shortener;
+    }
```

Comment:
I'm wondering that this could use FQCN as the ID when we consider autowiring, right?
```suggestion
    public function addService(ShortenerServiceInterface $shortener): void
    {
        $this->services[get_class($shortener)] = $shortener;
    }
```